server/lib/dataverse/provider.py

```diff
@@ -0,0 +1,246 @@
+import json
+import re
+import os
+from urllib.parse import urlparse, urlunparse, parse_qs
+from urllib.request import urlopen, Request
+from urllib.error import HTTPError
+
+from girder import events, logger
+from girder.models.setting import Setting
+
+from ..import_providers import ImportProvider
+from ..data_map import DataMap
+from ..file_map import FileMap
+from ..import_item import ImportItem
+from ..entity import Entity
+from ... import constants
+
+_DOI_REGEX = re.compile(r'(10.\d{4,9}/[-._;()/:A-Z0-9]+)', re.IGNORECASE)
+_QUOTES_REGEX = re.compile(r'"(.*)"')
+_CNTDISP_REGEX = re.compile(r'filename="(.*)"')
+
+
+def _query_dataverse(search_url):
+    resp = urlopen(search_url).read()
+    data = json.loads(resp.decode('utf-8'))['data']
+    if data['count_in_response'] != 1:
+        raise ValueError
+    item = data['items'][0]
+    files = [{
+        'filename': item['name'],
+        'mimeType': item['file_content_type'],
+        'filesize': item['size_in_bytes'],
+        'id': item['file_id'],
+        'doi': 'fixme'
+    }]
+    title = item['name']
+    title_search = _QUOTES_REGEX.search(item['dataset_citation'])
```

Comment:
Please feel free to open an issue at https://github.com/IQSS/dataverse/issues about exposing the title of a dataset separately from the citation in the Dataverse Search API.

Reply:
Done: https://github.com/IQSS/dataverse/issues/5339